src/common/libsubprocess/remote.c

```diff
@@ -438,6 +442,13 @@ static int remote_output_buffered (flux_subprocess_t *p,
     if (data && len) {
         int tmp;
 
+        /* In the event the buffer is full, the `fbuf_write()` will
+         * fail.  Call user callback to give them a chance to empty
+         * buffer.  If they don't, we'll hit error below.
+         */
+        if (!fbuf_space (c->read_buffer))
+            c->output_cb (c->p, c->name);
+
```

Comment:
Was going to comment that throwing an error here is not consistent with allowing the user to stop the stream from the callback, but I see the stream_start/stop functions are noted to be for local processes only.

We can fix that once we have credit based flow control since the remote will never send more data than we have room to put in the buffer.

Reply:
the callback placement here is unfortunate, the reason was the libev did not call things in the order I expected.   I expected:

```
output - put data in buffer, start output prep/check
check_cb - call output callback since there's data in the buffer
<start next iteration of libev loop>
```

but what happened was

```
output - put data in buffer, start output prep/check
<start next iteration of libev loop>
prep_check - see data in output buffer, start idle
output - want to put more data in buffer, hit EOVERFLOW
check_cb - this is never reached because of error above
```

the fact I just started the prep/check means check isn't called in the current iteration.

Reply:
> We can fix that once we have credit based flow control since the remote will never send more data than we have room to put in the buffer.

The work on #6291 is only for stdin since that's the specific case brought up by the user.  But yeah, for output we should add that as a todo as well.

Reply:
On the order of events, should the output watcher be stopped when the buffer is full then, and restarted when it's not?

Reply:
I think that's sort of like putting an infinite max size on the cbuf - it allows the sender to cause the receiver to use up an unchecked amount of memory.  Not great.

Another route might be to expose watcher priorities in our APIs, and then use them to ensure this check watcher runs before anything else.

But maybe the right solution is flow control.

Reply:
> Another route might be to expose watcher priorities in our APIs, and then use them to ensure this check watcher runs before anything else.

I wasn't aware of the libev priority stuff.  Hmmmm.  I suppose that could be an option, but at this point in time I'm not sure we have a way to add a priority to whatever underneath the covers is calling the flux future's then callback?  So perhaps this is something to simply kick the can down the road.

Reply:
Well we could elevate the priority of just this check watcher to get it to be called before the  check watcher in the future implementation.   Did you want to pause and try that?  I could give you a commit that adds a `flux_watcher_set_priority()` function to cherry pick.  Just as an experiment?

Reply:
Untested but here it is: 1932ce57c5c167385759e4fef7cd22294f5d25ca

Just call `flux_watcher_set_priority (check_watcher, 1);`

That should raise the priority from the default of 0 to 1.  if that doesn't work try 2 :-)  2 is the max.

Reply:
> Did you want to pause and try that?

As this would involve more than a few line tweak, I'm inclined to merge this PR and experiment with it in a different PR.  But lets log so we don't forget this conversation.

Reply:
Sure.